src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,286 @@
+        _setCalculationIntervalSeconds(_calculationIntervalSeconds);
+        _setGlobalOperatorCommission(_globalCommissionBips);
+    }
+
+    /**
+     * @notice Creates a new range payment on behalf of an AVS, to be split amongst the
+     * set of stakers delegated to operators who are registered to the AVS.
+     * @param rangePayments The range payments being created
+     */
+    function payForRange(
+        RangePayment[] calldata rangePayments
+    ) external onlyWhenNotPaused(PAUSED_PAY_FOR_RANGE) nonReentrant {
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentHash = _payForRange(rangePayment);
+            emit RangePaymentCreated(msg.sender, rangePaymentHash, rangePayment);
+        }
+    }
+
+    /**
+     * @notice similar to `payForRange` except the payment is split amongst *all* stakers
+     * rather than just those delegated to operators who are registered to a single avs
+     */
+    function payAllForRange(
+        RangePayment[] calldata rangePayments
+    ) external onlyWhenNotPaused(PAUSED_PAY_ALL_FOR_RANGE) onlyPayAllForRangeSubmitter {
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentHash = _payForRange(rangePayment);
+            emit RangePaymentForAllCreated(msg.sender, rangePaymentHash, rangePayment);
+        }
+    }
+
+    function _payForRange(RangePayment calldata rangePayment) internal returns (bytes32) {
+        require(rangePayment.strategiesAndMultlipliers.length > 0, "PaymentCoordinator.payForRange: no strategies set");
+        require(rangePayment.amount > 0, "PaymentCoordinator.payForRange: amount cannot be 0");
+        require(
+            rangePayment.duration <= MAX_PAYMENT_DURATION,
+            "PaymentCoordinator.payForRange: duration exceeds MAX_PAYMENT_DURATION"
+        );
+        require(
+            rangePayment.duration % calculationIntervalSeconds == 0,
+            "PaymentCoordinator.payForRange: duration must be a multiple of calculationIntervalSeconds"
+        );
+        // If retroactive payments enabled must at least be past LOWER_BOUND_START_RANGE,
+        // otherwise should start earliest at current block timestamp
+        require(
+            (retroactivePaymentsEnabled && rangePayment.startTimestamp >= LOWER_BOUND_START_RANGE) ||
+                rangePayment.startTimestamp >= block.timestamp,
+            "PaymentCoordinator.payForRange: invalid startTimestamp set"
+        );
+
+        // Require rangePayment is for whitelisted strategies
+        for (uint256 j = 0; j < rangePayment.strategiesAndMultlipliers.length; j++) {
+            IStrategy strategy = rangePayment.strategiesAndMultlipliers[j].strategy;
+            require(
+                strategyManager.strategyIsWhitelistedForDeposit(strategy),
```

Comment:
Could we determine if there is anything deposited in the strategy manager for that? Basing payments on what is whitelisted for deposit could change over time, or maybe deposits are disabled but there is still active delegation in the system?

Reply:
We could have a check for non zero value of `strategy.totalShares`. Imo I don't think removing certain strategies from whitelist will occur frequently if at all except for potential exploit vulnerabilities.

Reply:
Also adding beaconChainETHStrat check here https://github.com/Layr-Labs/eigenlayer-contracts/commit/a3aab21914ae5548b63100b39ecdec463212b9dd#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2R172

Can check it from the DM here https://github.com/Layr-Labs/eigenlayer-contracts/blob/dev/src/contracts/core/DelegationManager.sol#L38